query-engine/js-connectors/src/queryable.rs

```diff
@@ -167,10 +162,8 @@ impl JsQueryable {
             .map_err(into_quaint_error)?;
 
         let len = result_set.len();
-        let deserialization_span = info_span!("js:query:result", user_facing = true, "length" = %len);
-        Self::transform_result_set(self.flavour, result_set)
-            .instrument(deserialization_span)
-            .await
+        let _deserialization_span = info_span!("js:query:result", user_facing = true, "length" = %len).entered();
```

Comment:
Is this `.entered()` semantically equivalent to the previous `.instrument(deserialization_span)`?

Reply:
Yes. As far as I could tell, `transform_result_set` is async just so we could `instrument` it. In that case, we can just do `span!(..).entered()` and that would create a span from the moment `.entered()` called until the end of the block (function in this case)